add DeleteTaskCli , UpdatTaskCli

Farah-Dev4 opened the pull request.

Farah-Dev4 committed:
add DeleteTaskCli and DeleteTaskUseCaseTest
Modify on Exceptions.kt also

mostafa-n3ma committed:
Merge branch 'refs/heads/develop' into task/90-delete-task-cli

# Conflicts:
#	src/main/kotlin/domain/customeExceptions/Exceptions.kt
#	src/test/kotlin/domain/usecases/task/DeleteTaskUseCaseTest.kt

mostafa-n3ma committed:
update deleteTaskCli and deleteTaskUseCaseTest

AhmedMohamed056 commented on src/main/kotlin/presentation/cli/task/DeleteTaskCli.kt:
```diff
@@ -0,0 +1,42 @@
+
+
+class DeleteTaskCli (
+    private val getAllTaskStatesUseCase: GetAllTaskStatesUseCase,
```
use getAllTaskUseCase not getAllTaskStatesUseCase

Farah-Dev4 committed:
mdify some test on DeleteTaskCliTest

mostafa-n3ma commented on src/main/kotlin/presentation/cli/task/DeleteTaskCli.kt:
```diff
@@ -0,0 +1,42 @@
+            uiController.printMessage("${index + 1}. ${task.name}")
+
+        }
+        if (tasks.isEmpty()) throw NoTasksFoundException()
```
remove the through and handle it when deleting with the use case method

Farah-Dev4 committed:
add try and catch and do some modify

abdulazizacc commented on src/main/kotlin/presentation/cli/task/DeleteTaskCli.kt:
```diff
@@ -0,0 +1,76 @@
+import domain.entities.Task
+
+class DeleteTaskCli(
+    private val getAllTasksUseCase: GetAllTasksUseCase,
```
you must get your branch up to date so you can use all classes you need in your task

abdulazizacc commented on src/main/kotlin/presentation/cli/task/DeleteTaskCli.kt:
```diff
@@ -0,0 +1,76 @@
+        val selectedTask = tasks[index - 1]
+        uiController.printMessage(" Are you sure you want to delete: ${selectedTask.title}? (yes/no)")
+        val confirmation = uiController.readInput().trim().lowercase()
+
```
add message to check if it is not yes or no (invalid message) to try again, as you did with other scenarios before

abdulazizacc commented on src/main/kotlin/presentation/cli/task/DeleteTaskCli.kt:
```diff
@@ -0,0 +1,76 @@
+            } else {
+                uiController.printMessage(" Task not found.")
+            }
+
```
we should use try and here
`uiController.printMessage(" Task deleted.")` should be inside the try scope
and else should represent the catch

abdulazizacc commented on src/test/kotlin/presentation/cli/task/DeleteTaskCliTest.kt:
```diff
@@ -0,0 +1,73 @@
+            deleteTaskUseCase = deleteTaskUseCase,
+            uiController = uiController
+        )
+    }
```
here you do not invoke all the parameters of **DeleteTaskCli** class correctly.
and again update your branch so you can use the classes that you need.

abdulazizacc commented on src/main/kotlin/presentation/cli/task/DeleteTaskCli.kt:
```diff
@@ -0,0 +1,76 @@
+        if (currentUser == null) {
+            uiController.printMessage(" You are not logged in.")
+            return
+        }
```
you do not need to check for user authentication here

abdulazizacc reviewed (changes requested):
your code is already does not work, make your code run first before merging it

mostafa-n3ma committed:
Merge branch 'refs/heads/develop' into task/90-delete-task-cli

mostafa-n3ma committed:
get all tasks use case starting to test

mostafa-n3ma committed:
transforming get task grouped by state use case and get tasks use case  with tests

mostafa-n3ma committed:
updating entities

mostafa-n3ma committed:
updating and running fresh

mostafa-n3ma committed:
delete task cli with tests

mostafa-n3ma committed:
updating di with delete task cli dependencies

mostafa-n3ma committed:
adding update task cli class with tests
updating delete task cli class with tests
creating task cli util class with tests

mostafa-n3ma committed:
update presentation di with update task cli injection

MustafaIbrahim96 commented on src/main/kotlin/domain/repositories/TaskStateRepository.kt:
```diff
@@ -1,11 +1,12 @@
     fun createTaskState(state: TaskState): Boolean
     fun editTaskState(state: TaskState): Boolean
-    fun deleteTaskState(stateId: String): Boolean
+    fun deleteTaskState(stateId: UUID): Boolean
```
maybe you want project ID too to delete task from specific project

MustafaIbrahim96 commented on src/main/kotlin/presentation/cli/TaskState/DeleteTaskStateCli.kt:
```diff
@@ -16,7 +17,7 @@ class DeleteTaskStateCli(
         }
 
-        val result = deleteTaskStateUseCase.execute(taskId)
+        val result = deleteTaskStateUseCase.execute(UUID.randomUUID())
```
uuid.random that from my work and i will do it today dont worry

MustafaIbrahim96 reviewed (approved):
Greate job You can merge now allready solved failed test cases

MustafaIbrahim96 committed:
resolve test cases

MustafaIbrahim96 committed:
fix delete task state by only id task state
